src/admin-events/admin-events.js

```diff
@@ -0,0 +1,96 @@
+const { Finding, FindingSeverity, FindingType } = require('forta-agent');
+const { getAbi, getAddress } = require('@uma/contracts-node');
+const ethers = require('ethers');
+
+// load config files
+const { umaEverestId } = require('../../agent-config.json');
+const adminEvents = require('./admin-events.json');
+
+// returns the list of events for a given contract
+function getEvents(contractName) {
+  const events = adminEvents[contractName];
+  if (events === undefined) {
+    return []; // no events for this contract
+  }
+  return events;
+}
+
+// get contract names for mapping to events
+const contractNames = Object.keys(adminEvents);
+
+// Create the interfaces for each contract that has events we wish to monitor
+const ifaces = {};
+contractNames.forEach((contractName) => {
+  // Get the abi for the contract
+  const abi = getAbi(contractName);
+
+  // create ethers interface object
+  const iface = new ethers.utils.Interface(abi);
+
+  // Create an association between the contract name and the interface
+  ifaces[contractName] = iface;
+});
+
+// Filters the logs to only events in eventNames
+function filterAndParseLogs(logs, address, iface, eventNames) {
+  // collect logs only from the contract
+  const contractLogs = logs.filter((log) => log.address === address);
```

Comment:
Seems like the function shouldn't continue past this point if the address doesn't match--I would return from the function  immediately if contractLogs is empty.

Reply:
Done